supportedBackends/aws/src/main/scala/cromwell/backend/impl/aws/AwsBatchAsyncBackendJobExecutionActor.scala

```diff
@@ -0,0 +1,499 @@
+
+  // TODO: Determine the necessity for this monitoring stuff
+  lazy val monitoringScript: Option[AwsBatchInput] = {
+    // callPaths.workflowPaths.monitoringScriptPath map { path =>
+    //   AwsBatchFileInput(s"$monitoringParamName-in", path.pathAsString, localMonitoringScriptPath, workingDisk)
+    // }
+    None
+  }
+
+  lazy val monitoringOutput: Option[AwsBatchFileOutput] = monitoringScript map { _ =>
+    AwsBatchFileOutput(s"$monitoringParamName-out",
+      callPaths.monitoringLogPath.pathAsString, localMonitoringLogPath, workingDisk)
+  }
+
+  override lazy val commandDirectory: Path = AwsBatchWorkingDisk.MountPoint
+
+  private val DockerMonitoringLogPath: Path = AwsBatchWorkingDisk.MountPoint.resolve(callPaths.monitoringLogFilename)
+  private val DockerMonitoringScriptPath: Path = AwsBatchWorkingDisk.MountPoint.resolve(callPaths.monitoringScriptFilename)
+
+  override def scriptPreamble: String = {
+    if (monitoringOutput.isDefined) {
+      s"""|touch $DockerMonitoringLogPath
+          |chmod u+x $DockerMonitoringScriptPath
+          |$DockerMonitoringScriptPath > $DockerMonitoringLogPath &""".stripMargin
+    } else ""
+  }
+
+  override def globParentDirectory(womGlobFile: WomGlobFile): Path = {
+    val (_, disk) = relativePathAndVolume(womGlobFile.value, runtimeAttributes.disks)
+    disk.mountPoint
+  }
+
+  override def isTerminal(runStatus: RunStatus): Boolean = {
+    runStatus match {
+      case _: TerminalRunStatus => true
+      case _ => false
+    }
+  }
+
+  // Primary entry point for cromwell to actually run something
+  override def executeAsync(): Future[ExecutionHandle] = {
+    for {
+      submitJobResponse <- Future.fromTry(batchJob.submitJob())
+    } yield PendingExecutionHandle(jobDescriptor, StandardAsyncJob(submitJobResponse.jobId), Option(batchJob), previousStatus = None)
+  }
+
+  val futureKvJobKey = KvJobKey(jobDescriptor.key.call.fullyQualifiedName, jobDescriptor.key.index, jobDescriptor.key.attempt + 1)
+
+  // This is called by Cromwell after initial execution (see executeAsync above)
+  // It expects a Future[RunStatus]. In this case we'll simply call the
+  // AWS Batch API to do this. The AwsBatchJob object in the PendingExecutionHandle
+  // will have the actual status call, so our job here is simply to pull the
+  // object out of the handle and execute the underlying status method
+  override def pollStatusAsync(handle: AwsBatchPendingExecutionHandle): Future[RunStatus] = {
+    val jobId = handle.pendingJob.jobId
+    val job = handle.runInfo match {
+      case Some(job) => job
+      case None => throw new RuntimeException(s"pollStatusAsync called but job not available. This should not happen. Jobid ${jobId}")
+    }
+    val status = job.status(jobId)
+    status match {
+      case Success(runstatus) => { runstatus match {
+          case _: TerminalRunStatus => writeToPaths(jobId, job)
+          case _ => ()
+        }
+      }
+      case _ => ()
+    }
```

Comment:
```
status collect { case _: TerminalRunStatus => writeToPaths(jobId, job)}
```

Reply:
Clearly I have a blind spot for using collect. ;-) Thanks again. 604f3e4